neuralpp/symbolic/expression.py

```diff
@@ -74,15 +112,41 @@ def _convert(cls, from_expression: Expression) -> Expression:
         if isinstance(from_expression, cls):
             return from_expression
         match from_expression:
-            case Constant(value=value):
-                return cls.new_constant(value)
-            case Variable(name=name):
-                return cls.new_variable(name)
-            case FunctionApplication(function=function, arguments=arguments):
+            case Constant(value=value, type=constant_type):
+                return cls.new_constant(value, constant_type)
+            case Variable(name=name, type=variable_type):
+                return cls.new_variable(name, variable_type)
+            case FunctionApplication(function=function, arguments=arguments, type=function_type):
                 return cls.new_function_application(function, arguments)
             case _:
                 raise ValueError(f"invalid from_expression {from_expression}: {type(from_expression)}")
 
+    @property
+    def type(self) -> Expression:
+        return self._type
+
+    def get_function_type(self) -> FunctionType | Type:
+        """ Call me when self is a function """
+        match self:
+            case Constant(type=type_expression) | Variable(type=type_expression):
+                return type_expression.get_function_type_of_type_expression()
+            case FunctionApplication(function=function, number_of_arguments=num):
+                return function.get_return_type(num)
+
+    def get_function_type_of_type_expression(self) -> FunctionType | Type:
+        """ Call me when self is a type Expression of a function """
```

Comment:
Using "call me" for the function documentation (1st person for the function) is clear, but unusual and for that reason may be distracting. Can you please switch to a 3rd person description?

Reply:
yes